rust/ql/src/queries/unusedentities/UnusedVariable.qll

```diff
@@ -7,8 +7,12 @@ class DiscardVariable extends Variable {
 
 /** Holds if variable `v` is unused. */
 predicate isUnused(Variable v) {
+  // variable is accessed or initialized
   not exists(v.getAnAccess()) and
   not exists(v.getInitializer()) and
+  // variable is intentionally unused
   not v instanceof DiscardVariable and
-  not v.getPat().isInMacroExpansion()
+  // variable is in a context where is may not have a use
+  not v.getPat().isInMacroExpansion() and
+  not exists(FnPtrType fp | fp.getParamList().getParam(_).getPat() = v.getPat())
```

Comment:
I think it would be better to make this exclusion in `VariableImpl.qll`, just like we [do for traits](https://github.com/github/codeql/blob/6054855edbe68e2f73f9bac51e676a8165fada02/rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll#L90).

Reply:
Done.  Seems like a great idea, but I'll kick off another DCA run to be sure there are no unintended consequences...